src/scanner_v2.cpp

```diff
@@ -111,10 +112,10 @@ void ScannerV2::scannerStartedCB()
   PSENSCAN_INFO("ScannerController", "Scanner started successfully.");
   stopStartWatchdog();
   PSENSCAN_DEBUG("Scanner", "Inform user that scanner start is finsihed.");
-  scanner_has_started_.set_value();
+  std::for_each(scanner_has_started_.begin(), scanner_has_started_.end(), [](auto& promise) { promise.set_value(); });
 
   // Reinitialize
-  scanner_has_started_ = std::promise<void>();
+  scanner_has_started_.clear();
```

Comment:
Due to the fact that the `ScannerV2::start()` and the `ScannerV2::scannerStartedCB` functions are called from at least two different threads, namely the user-main thread and the `io_service` thread form the `UdpClient`, the promise vector needs to be protected against concurrent access via mutex or something.
If the promise vector is protected via mutex, one actually can skip the vector usage and continue using only one promise protected via mutex (or realiased via atomic).

Reply:
Note: `std::promise` cannot be used in combination with `std::atomic` because:
```
error: static assertion failed: std::atomic requires a trivially copyable type
       static_assert(__is_trivially_copyable(_Tp),
```